### ISSUE TITLE
Spanish language is not very usable

### ISSUE THREAD

**Issue by ngosang** [morpheus65535/bazarr#1986]
**Describe the bug**
"Spanish" and "Latin American Spanish" is the same language and 99.5 % of the words are the same. The differences between them are more or less the same differences between English from UK and US. There are small differences in pronunciation of some words but they are written in the same way. There is not Wikipedia in "Latin American Spanish", just one for Spanish.

Said that, I don't care about the differences we have, I just want subtitles in Spanish (any of them) and I can't achieve that in Bazarr because I have to choose one of theme. It's hard to find any of them in most series, try to find both is impossible. I tried with the "cutoff" and other options but I can't get what I want. Anyway, it shouldn't be so complicate to configure one of the most spoken languages in the world.

It's possible to merge both languages into Spanish?
If not, it's possible to configure a "fallback" language? For example, I want "Spanish" but if it's not found I want "Latin American Spanish".

I know this not qualifies as a software bug but the usability problem is so big that it makes the tool useless.

Real use case:
I have configured the series "Fullmetal Alchemist: Brotherhood" with "English" and "Spanish" subtitles. English is fine for all episodes. Half of the episodes only have subtitles in "Spanish", the other half only in "Latin American Spanish"  and some of them in none. Since I have configured "Spanish" half of the subtitles are auto-downloaded and I have to perform manual search + download for the other 30 episodes in "Latin American Spanish".

**Comment by ngosang** [comment]
I tried but I have english too. It will only download english and stops there.

**Comment by ngosang** [comment]
I am a native speaker and I have opened the issue because it is not possible to configure it in a way that is useful for most.
If it's not going to be fixed you can close it but I'm upset. It is the typical case in which a noisy minority creates a problem for the silent majority for ideological reasons. If there is only one Spanish dictionary and one Wikipedia, I don't know why we have different subtitles (they are copy and paste). The only explanation is hate and racism.
If you can improve the fallback or cutoff functionality in future versions I would appreciate it.

**Comment by morpheus65535** [comment]
I understand how you feel and appreciate your feedback. Right now, you're the first one to bring this to me. We also have the same differentiation for Portuguese and Chinese and it's been well received by users. I don't think it's a different case here.

From my perspective, I see that having both Spanish and English cause an issue where cutoff prevent any Spanish subtitles from being downloaded once English as been. I would suggest we try to find solutions to have a better cutoff system. Do you agree with me?

Now that we've said that an knowing you're a Python developper yourself, do you have any idea how we could achieve that? I'm open to suggestion. :-)

**Comment by ngosang** [comment]
> I would suggest we try to find solutions to have a better cutoff system. Do you agree with me?

Yes.

> Now that we've said that an knowing you're a Python developper yourself, do you have any idea how we could achieve that? I'm open to suggestion. :-)

The cutoff expression will be too complex to be configured or even displayed in the Web UI. It must be `'en' and ('es' or 'ea')` to fulfill the requirements.

A better alternative will be to define a fallback language. Something like this and keep the cutoff empty. This could be used for other languages too and it's more intuitive than the cutoff IMHO.
![imagen](https://user-images.githubusercontent.com/10577978/201762544-92fa6b8f-8b77-400b-a83b-b6457e14f21a.png)

If you don't like I can think something else in the weekend.

**Comment by sambartik** [comment]
I would suggest an alternative and more flexible approach. 

Let the cutoff operate on LanguageGroup entities rather than just bare Languages. Internally even a single Language would be a part of LanguageGroup with only one Language member. Then people with similar use case as ngosang could in the UI add two languages: Spanish and Latin American Spanish and group them together to a single LanguageGroup.

Each LanguageGroup would then have a property whether it is optional. Let's imagine we have the following language profile settings:
```
--- LanguageGroup (1)
           -- English
--- LanguageGroup (2)
           -- Spanish
           -- Latin American Spanish
--- LanguageGroup (3) [OPTIONAL]
           -- Mandarin
```

The flow would look like this:

1. Go through all LanguageGroups that were NOT marked as optional: LanguageGroup (1) and LanguageGroup (2)
    Then for each of the above defined LanguageGroup:
    I. Check if ANY of the Language members of the group have subtitles downloaded
    II. If yes: continue to next group
    III. Else: trigger sequential search for subtitles in the Languages defined in the group. The order they would be searched should be the same as the order they were defined in the UI.
2. If all LanguageGroups that were NOT marked as optional have their subtitles downloaded, then we're done.
3. If some LanguageGroups that were NOT marked as optional are missing their subtitles, then continue to search for subtitles in similar manner for the rest of LanguageGroups that were optional.

In our example, we achieved to require to download subtitles for LanguageGroup (1) and LanguageGroup (2). We require subtitles in english language and then in one of the "Spanish languages". If either of those LanguageGroups do not have any of their Languages downloaded, then we proceed to search also for all the other LanguageGroups that were marked as optional: LanguageGroup (3). In our example, it was mandarin language.

I am not sure whether this catches all possible use cases, so this suggestion might need additional thought, however I think the LanguageGroups idea is a step in the right direction.

**Comment by ngosang** [comment]
I was waiting for you. Thank you for ruining this software, for the class about my own language and for the proposal you made for improving the software. I want to use both spanish but I can't because you don't want to use mine.

**Comment by morpheus65535** [comment]
@vitiko98 @ngosang ok let put an end to that. We'll focus on improving Bazarr. If you wanna argue about how you live witheach other, do it somewhere else please.

**Comment by GermanG** [comment]
FYI 
I brought this issue in reddit
https://www.reddit.com/r/bazarr/comments/pmdozi/cutoff_and_es_vs_esmx

There's no easy way out, the ideas I had were:
- Make a kind of "language group" cutoff.
- Create "language synonyms" so bazaar can treat them as the same language.

Nothing can be resolved without messing deeply with bazaar.

So far I have "Spanish cutoff" or "any cutoff".

**Comment by ngosang** [comment]
I will try to be constructive from here.

Use cases:
* The user only wants Spanish MX subtitles. It's already supported
* The user only wants Spanish (Any) subtitles. It can be achieved with "cutoff".
* The user wants English and Spanish MX subtitles. It's already supported
* **The user wants English and Spanish (Any) subtitles. `'en' and ('es' or 'es-MX')` It's not supported yet.**

The requirements are:
* Spanish MX and Spanish Castilian are different languages so they must be separate
* Some users don't care about the differences and they prefer to use any of them (with or without priorities). The important thing is to find at least one Spanish subtitle
* It should be easy to configure in the Web UI
* It should be easy to implement in the current Bazarr code

Proposals until now:
* Add more complex expressions for the "cutoff" feature. To archive the requirements the user will have to write complex queries =>  `'en' and ('es' or 'es-MX')`
* Language fallback. See my previous comment for more information. It won't cover future corner case like 2 fallback languages...
* Language group. The user will be able to create a "virtual" language with one or more languages. They will be treated as a new language and the user can use it anywhere in Bazarr including the "cutoff".
* Language synonyms. It's similar to the "language group" solution but it will be easy to configure by the user. If you configure Spanish MX == Spanish Castilian, you can use any of them and Bazarr with handle it transparently.

IMHO the more complete option is the "language group" but I think it will require too many changes in the code. I also have some doubts:
* What will be the subtitle extension for the language group?
* We didn't define the priorities inside the group yet
* Does it make sense to let the user mix unrelated languages in the same group?

Since I see the task is too complex to implement in a sort period of time, a simplified "language group" solution will work for me:
* Bazarr could provide a "hardcoded / preconfigured" list of language groups that make sense. For example "Spanish (Any) = (Spanish MX or Spanish Castilian). Same for Portuguese, Chinese...
* The user could select "Spanish (Any)" anywhere in the Bazarr UI and it will work as expected.
* This solution will work for me, but some users will complain they prefer Spanish Castilian over Spanish MX when available.

@morpheus65535 Are any of this options doable?

**Comment by Magikarplvl4** [comment]
Maybe add a combined language as a third option? 
Something like Spanish (combined)

**Comment by rmichelena** [comment]
I totally agree with the need to be able to, as a user choice, consider both variants as synonyms, and even better (in case both were available) order them in priority. There's the even worse case of some subs sites coding as "es-MX" or "es-AR" or such separately, and I'd argue there's as much or more difference between Mexican and Argentinian variations than any of those against Spain's (Castilian) variation.
I guess the quick&dirty solution is ngosang's last suggestion, adding some "meta-languages" (in this case "Spanish(Any)" to the language choices, and having those metalanguages accept any variant. At least for Spanish it should be easy, you'd just search for "es*" and that'd pull up "es", "es-MX", and all those variants.

**Comment by albino1** [comment]
Please add an option for `Spanish (Any)`.  The current situation is really unusable because of all the micro-labeling.

Imagine an American failing to find `English` subs because the options were only labeled like `en-GB` or `en-CA`.

Would you really reject English subs if `color` was spelled `colour`?

**Comment by vitiko98** [comment]
#2135

**Comment by vitiko98** [comment]
![image](https://github.com/morpheus65535/bazarr/assets/59455966/f6472a3a-25f2-4d17-9df8-00013bf21d7b)

Here's the usage example (Settings/Languages). No big documentation yet, but I think the idea is clear.

This feature will be available in upcoming beta.

**Comment by ngosang** [comment]
@vitiko98 @morpheus65535 Thank you very much! I just tested this feature in Bazarr 1.2.2 and it's awesome.

The configuration to get subtitles in English + Spanish (Spain or Latino) is this:
![image](https://github.com/morpheus65535/bazarr/assets/10577978/7e6f8c52-39b0-47c0-9fd4-6bdaeee09d4e)

I think the first 2 "equals" aren't necessary because I have the following configuration in the profile. Anyway it's working.
![image](https://github.com/morpheus65535/bazarr/assets/10577978/22835d66-d74c-4924-8984-e951b7a4aeca)

**Comment by bejaranoo** [comment]
This is getting ridiculous but for lacking of official explanation on this function, I gotta make sure
![image](https://github.com/morpheus65535/bazarr/assets/67523369/7d5405da-3b19-4c7d-9e32-a0aafb6d59ea)

**Comment by morpheus65535** [comment]
> This is getting ridiculous but for lacking of official explanation on this function, I gotta make sure
> 
> ![image](https://github.com/morpheus65535/bazarr/assets/67523369/7d5405da-3b19-4c7d-9e32-a0aafb6d59ea)
> 
> 

If you need configuration support, you can reach support team or community member on Discord. 

Regarding the documentation issue, we gladly accept PR to improve it.

**Comment by rdmartins** [comment]
Do this configuration considers embedded subtitles and audio language?

I ripped my Friends DVD collection with the Brazilian Portuguese subtitles embedded.
I understand that Bazarr will consider them as European Portuguese, even if it is Brazilian.

Also, I have a Brazilian series, with Brazilian Portuguese audio, and I understand that Bazarr will also consider it European Portuguese audio.

I do the configuration to threat European Portuguese as Brazilian Portuguese, but Bazarr still wants to download Brazilian Portuguese subtitles :(

My config (I tried many)
![image](https://github.com/morpheus65535/bazarr/assets/13852370/fbe9dfb6-becc-426d-85a7-dd3cb2fcdc82)
![image](https://github.com/morpheus65535/bazarr/assets/13852370/e626de7b-42f8-44c1-843b-4d8fe29847fd)
My embedded issue:
![image](https://github.com/morpheus65535/bazarr/assets/13852370/fdd8e509-032b-48f5-a4ff-1a5463e8a5fe)
My audio issue:
![image](https://github.com/morpheus65535/bazarr/assets/13852370/87501765-bf12-498d-814d-83eab1fc5faf)

**Comment by dantebarba** [comment]
TL;DR below is my configuration for bazarr to work with both ES-MX and ES, and EN. I'm unable to make it cutoff with ES or ES-MX

<img width="1070" alt="Image" src="https://github.com/user-attachments/assets/d7cb70d7-fa47-4a6c-babc-8ababe6907ff" />

<img width="702" alt="Image" src="https://github.com/user-attachments/assets/f21a1ee9-7a20-4ad2-8d8a-0972c3771c54" />

### My problem with Spanish subtitules

Putting aside all the controversy with Spanish language variations, [ngosang](https://github.com/ngosang) is right, splitting Spanish between Spanish (MX) and Castilian Spanish generates some issues for people who wants to configure Spanish + English profiles.

In my case I'd like to always download English + Spanish subtitles but:

1. Spanish subtitles should be mandatory. I WANT them
2. English subtitules are optional. I'd LIKE to have them but if I can't find them it's ok.
3. In an ideal world I will always choose Spanish MX over Spanish because it resembles the experience to Netflix for us Latin Americans.
4. But downloading subtitles is not ideal, so I'd rather prefer to always have te BEST Spanish subtitle. This means the one that can actually be found and that has the best score. It happened to me a lot lately that some movies do have ES-MX subtitles but do not have an ES alternative. There are other cases were both ES and ES-MX are available, but one has a 90% score and the other one has a 80% score. 

Take a look at this case

<img width="1077" alt="Image" src="https://github.com/user-attachments/assets/574fb5f3-163a-4595-b723-a52d0b1133b3" />

The subtitle for babygirl is not available in ES but it was downloaded in ES-MX. In this case I'm fine cutting off at ES-MX. 

I know that the ideal solution would be having `LanguageGroups`. Also another option could be having multiple profiles assigned instead of just one. In that case I would have `English` with `cutoff`: any and `Spanish` with cutoff: `any`

But if none of those options are feasible, just adding an advanced option for cutoff and using regex to match the language string would be great. The regex could be applied to a concatenated string with all the available languages and if it matches, then the cutoff has been met.

**Comment by dantebarba** [comment]
Related to https://github.com/morpheus65535/bazarr/issues/2820
